### PR TITLE
README.md: Add usage and demo sections

### PR DESCRIPTION
**Pull request by Abijeet**
Bug: https://phabricator.wikimedia.org/T379709

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,6 +53,46 @@ banana-i18n is a javascript internationalization library that uses "banana" form
...
+console.log(banana.i18n('greet-user', 'Alice')); // Output: Bonjour, Alice!
+```
+
+### JavaScript
+
+For usage in JavaScript, please check the files under the `demo` folder after building the distribution files by running `npm run build`.
+
+## Demo
+
+See the library in action: https://wikimedia.github.io/banana-i18n/demo/
```
Comment (hueitan) [inline README.md:93]: the demo page shows nothing to me, is this normal?
image

Comment (Abijeet) [reply README.md:93]: Its trying to load the library from the dist folder but we don't push the files there. We have two options:

1. Remove the dist folder from `gitignore` and add the files from the build into the dist folder.
2. Add the build files separately under the demo.

Comment (hueitan) [reply README.md:93]: 2. You can do it through the CI and only build and store it on gh-pages branch

Comment (Abijeet) [reply README.md:93]: @hueitan That was something I was considering. I assume that would be easy to setup?

Comment (hueitan) [reply README.md:93]: > @hueitan That was something I was considering. I assume that would be easy to setup?

yes, we did that in wikipedia-preview, https://github.com/wikimedia/wikipedia-preview/blob/9d775e477919f4c3b17fb126685433e72baa2784/.github/workflows/main.yml#L26-L52, but through the GitHub Action.

Comment (santhoshtr) [reply README.md:93]: gh-pages branch approach seems better to me

Comment (hueitan) [reply README.md:93]: the demo page is available with this CI update https://github.com/wikimedia/banana-i18n/pull/88

**Review by hueitan — APPROVED** [review]
LGTM

**Review by Nikerabbit — APPROVED** [review]

**Pull request merged**